src/consumed.py

```diff
@@ -454,18 +459,23 @@ def build_matrices(self, pol: str, adj: str, date):
             else:
                 G[i] = self.generation.loc[date, r]
 
+        E = np.nan_to_num(E)
+        G = np.nan_to_num(G)
+        ID = np.nan_to_num(ID)
+
         # In some cases, we have zero generation but non-zero transmission
         # usually due to imputed zeros during physics-based cleaning being set to 1.0
-        # but sometimes due to ok values being set to 1.0ß
+        # but sometimes due to ok values being set to 1.0
         to_fix = (ID.sum(axis=1) > 0) & (G == 0)
         ID[:, to_fix] = 0
         ID[to_fix, :] = 0
```

Comment:
So in https://github.com/singularity-energy/open-grid-emissions/pull/259#issuecomment-1364105527 you mentioned re-writing this check to fix the HGMA issue but it looks like nothing has changed here - it there another commit you need to push? I'm not sure that I'm clearly seeing in these proposed changes what is fixing the HGMA issue

Reply:
Fixed by moving nan to zero conversion above this ID / Gen check